### ISSUE TITLE
create separate rule to match socket connects

### ISSUE THREAD

**Issue by mike-hunhoff** [mandiant/capa-rules#965]
capa presently only matches connecting TCP sockets, but it should match all connects as this is usually of interest. At a minimum, we should add a new rule to detect connecting UDP sockets.

https://github.com/mandiant/capa-rules/blob/e50d824553ae8d7168ac2d1f7b5540a93f944e28/communication/socket/tcp/connect-tcp-socket.yml#L19-L24

**Comment by hafizfarhad** [comment]
Hi @mike-hunhoff,  

I’d like to work on this issue. Could you please assign it to me?  

My plan is to mirror the structure of the TCP rule and create a new rule (`connect-udp-socket.yml`) to detect UDP connections by adding relevant APIs like `sendto`, `recvfrom`, `WSASendTo`, and `WSARecvFrom`.  

Would this approach work? Or do you have any specific recommendations? 

Thanks!

**Comment by mike-hunhoff** [comment]
Great, I've assigned you this issue. We aim to reuse as much as possible with our capa rules. One approach here would be to create a new, more generic rule named something like `connect socket`. From there, modify `connect tcp socket` and create a new rule named `connect udp socket`, where the former contains only `match(create TCP socket)` and `match(connect socket)` and the latter contains only `match(create UDP socket)` and `match(connect socket)` for features. This enables us to reuse the logic found in `connect socket` for both rules. Please reach out here if you have specific questions along the way and don't forget to review our [extensive rule format documentation](https://github.com/mandiant/capa-rules/blob/master/doc/format.md) if you haven't already 😄

**Comment by hafizfarhad** [comment]
Hi Sir,

I have submitted a PR and hope it works as expected. Since this is my first PR, I’d really appreciate any feedback or suggestions if there are mistakes.

I also ran linting on my changes and encountered an error because I hadn’t provided an example in the `create UDP socket` rule. I couldn’t find any examples in the test files, so I created a simple `C` file that uses `connect()` to trigger the rule. Should I upload this file into the [test files](https://github.com/mandiant/capa-testfiles) and add _examples_ in the `create UDP socket` rule, or is it acceptable to leave it out?

Thank you for your time and guidance!

**Comment by mike-hunhoff** [comment]
Yes, please create a PR at https://github.com/mandiant/capa-testfiles for your test file, thank you!
